006-adjusting-to-support-heroku.md

```diff
@@ -0,0 +1,126 @@
+# Adjusting Appwrite to support Heroku <!-- What do you want to call your `awesome_feature`? -->
+
+- Implementation Owner: @lohanidamodar
+- Start Date: 12-01-2021
+- Target Date: Unknown
+- Appwrite Issue:
+  https://github.com/appwrite/appwrite/issues/547
+
+## Summary
+
+[summary]: #summary
+
+<!-- Brief explanation of the proposed contribution. Write your answer below. -->
+
+Heroku and similar platforms are modern day standards for deploying applications. But deploying on such platforms requires certain fundamental architectural decisions made. Supporting these platforms will help make deploying and testing Appwrite easier for beginners. We can even support one click deployment right from Github.
+
+## Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+**What problem are you trying to solve?**
+
+<!-- Write your answer below. -->
+Heroku, deploys applications on an **ephemeral** storage called dynos, where data can be written to disk however those data will not persist after the application is restarted. This provides fundamental challenge to deploy Appwrite, as appwrite uses filesystem storaging and managing things like uploads, functions, config and cache.
+
+Next up, for services like database and redis, there are Heroku Add-ons that can be used, however there is no influxdb/telegraf Add-ons on Heroku.
+
+These problems are limiting the ability of Appwrite to be deployed in Heroku
+
+
+**What is the context or background in which this problem exists?**
+
+<!-- Write your answer below. -->
+
+The current implementation of storage and influxdb/telegraf prevents from deployment to Heroku and make it usable properly.
+
+**Once the proposal is implemented, how will the system change?**
+
+<!-- Write your answer below. -->
+
+- There will be an option to disable influxdb/telegraf stats using environment variable.
+- There will be an option to switch default storages (uploads, configs, cache and functions) to external services like Amazon S3 or Digitalocean Spaces (may be).
+
+<!-- Please avoid discussing your proposed solution. -->
+
+## Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+### Introduce new environment variable to disable stats
+Should introduce a new environment variable _APP_STATS, default value is **enabled** and can be set to **disabled** to disable the stats, enabling user to completely drop influxdb/telegraph without any errors
```

Comment:
Not 100% sure about the env name, we can discuss that in the separate RFC for disabling usage stats.

Reply:
Don't forget to mention in the docs that using this option, devs should disable both influx, telegraf and the usage worker.